group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1351,6 +1355,7 @@ private void throwIfShareGroupHeartbeatRequestIsInvalid(
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
                 throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
             }
+            throwIfEmptyString(request.memberId(), "MemberId can't be empty.");
```

Comment:
Since `ShareGroupHeartbeat RPC` is still under development, I assume this change is acceptable without a version check.

Reply:
Since consumers are now required to generate their own `memberId` and the `ShareGroupHeartbeat RPC` is still under development, it seems feasible to check the `memberId` at the beginning: 
```java
    private void throwIfShareGroupHeartbeatRequestIsInvalid(
        ShareGroupHeartbeatRequestData request
    ) throws InvalidRequestException, UnsupportedAssignorException {
        throwIfEmptyString(request.groupId(), "GroupId can't be empty.");
        throwIfEmptyString(request.rackId(), "RackId can't be empty.");
        throwIfEmptyString(request.memberId(), "MemberId can't be empty.");

         if (request.memberEpoch() == 0) {
            if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
                throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
            }
        } else {
            throw new InvalidRequestException("MemberEpoch is invalid.");
        }
    }

Dose this make sense ?
```

Reply:
Since consumers are now required to generate their own `memberId` and the `ShareGroupHeartbeat RPC` is still under development, it seems feasible to check the `memberId` at the beginning: 
```java
private void throwIfShareGroupHeartbeatRequestIsInvalid(
    ShareGroupHeartbeatRequestData request
) throws InvalidRequestException, UnsupportedAssignorException {
    throwIfEmptyString(request.groupId(), "GroupId can't be empty.");
    throwIfEmptyString(request.rackId(), "RackId can't be empty.");
    throwIfEmptyString(request.memberId(), "MemberId can't be empty."); // Always check memberId

    if (request.memberEpoch() == 0) {
        if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
            throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
        }
    } else if (request.memberEpoch() < ShareGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH) {
        throw new InvalidRequestException("MemberEpoch is invalid.");
    }
}

```
Dose this make sense ?